api/v1alpha1/ocivalidator_types.go

```diff
@@ -17,17 +17,65 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"strings"
+
 	validatorv1alpha1 "github.com/spectrocloud-labs/validator/api/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // OciValidatorSpec defines the desired state of OciValidator
 type OciValidatorSpec struct {
-	RegistryAuthRules []RegistryAuthRule `json:"registryAuthRules,omitempty" yaml:"registryAuthRules,omitempty"`
+	OciRegistryRules []OciRegistryRule `json:"ociRegistryRules,omitempty" yaml:"ociRegistryRules,omitempty"`
+	EcrRegistryRules []EcrRegistryRule `json:"ecrRegistryRules,omitempty" yaml:"ecrRegistryRules,omitempty"`
+}
+
+func (s OciValidatorSpec) ResultCount() int {
+	return len(s.EcrRegistryRules) + len(s.OciRegistryRules)
+}
+
+type OciRegistryRule struct {
+	Host           string    `json:"host" yaml:"host"`
+	RepositoryPath string    `json:"repositoryPath,omitempty" yaml:"repositoryPath,omitempty"`
+	Auth           BasicAuth `json:"auth,omitempty" yaml:"auth,omitempty"`
+	Cert           string    `json:"cert,omitempty" yaml:"cert,omitempty"`
+}
+
+func (r OciRegistryRule) Name() string {
+	var sb strings.Builder
+	sb.WriteString(r.Host)
+	sb.WriteString("/")
+	sb.WriteString(r.RepositoryPath)
+	return sb.String()
+}
+
+type BasicAuth struct {
+	Username string `json:"username" yaml:"username"`
+	Password string `json:"password" yaml:"password"`
+}
+
+type EcrRegistryRule struct {
+	Host           string  `json:"host" yaml:"host"`
+	RepositoryPath string  `json:"repositoryPath,omitempty" yaml:"repositoryPath,omitempty"`
+	Auth           EcrAuth `json:"auth,omitempty" yaml:"auth,omitempty"`
+	Cert           string  `json:"cert,omitempty" yaml:"cert,omitempty"`
```

Comment:
`Cert` may not be needed for ECR since the endpoints are public / managed by AWS and should use a publicly verifiable CA cert.

Reply:
Sounds good, I'll remove it for now. Main reason I left it in is because the UI is setup in a way that allows you to pass in a cert even for ECR

Reply:
Oh.. interesting. Maybe you can configure a custom cert for ECR with private endpoints.. I didn't dig too deeply in the AWS docs.

Reply:
I just tried configuring a private ECR repo in the console and couldn't find any options pertaining to certs. Let's leave that field out for now and assume our UI is incorrect.